src/Symfony/Component/HttpFoundation/HeaderBag.php

```diff
@@ -105,23 +105,40 @@ public function add(array $headers)
      * @param string|null $default The default value
      * @param bool        $first   Whether to return the first value or all header values
      *
-     * @return string|string[]|null The first header value or default value if $first is true, an array of values otherwise
+     * @return string|string[]|int|null The first header value or default value if $first is true, an array of values otherwise
      */
     public function get($key, $default = null, $first = true)
+    {
+        $all = $this->getValues($key, $default);
+
+        if (true === $first) {
+
+            if (\count($all)) {
+                return isset($all[0]) ? $all[0] : $default;
+            }
+
+            return $default;
+        }
+
+        @trigger_error(sprintf('Passing $first as a third argument is deprecated since Symfony 4.4 and will be removed in Symfony 5.0. %s will return the first header value, use %s::gets() instead to get an array of values.',  __METHOD__, __CLASS__), E_USER_DEPRECATED);
+
+        return $all;
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getValues($key, ?string $default = null): array
```

Comment:
why does `$key` not have a string type? also a string default value for an array return value does not make any sense to me. There shouldn't be a default value IMO and it can just return an empty array if the header doesn't exist.

Reply:
The default should be here because else we break the compatibility, we always return the default as an array. It should be removed when $first is removed.